src/proto/h1/dispatch.rs

```diff
@@ -233,15 +233,6 @@ where
     }
 
     fn poll_read_head(&mut self, cx: &mut task::Context<'_>) -> Poll<crate::Result<()>> {
-        // can dispatch receive, or does it still care about, an incoming message?
-        match ready!(self.dispatch.poll_ready(cx)) {
```

Comment:
Sorry, I didn't notice this needs to be reverted too, otherwise we never actually check the client's `poll_ready` 🙈 .

Everything else looks perfect (I was just about to merge, and notice this.)

Reply:
Np, good/important catch. TBH, I've been relying on you for soundness ever since CR process started.

Reverted, I also changed the wording of the comment a bit, it didn't sound right to me. It's now:
```
// can dispatch receive, or does it still care about other incoming message?
```